starfish/test/image/test_imagestack_labeled_indices.py

```diff
@@ -0,0 +1,206 @@
+NUM_Z = len(Z_LABELS)
+
+
+def fill_value(round_: int, ch: int, z: int) -> float:
+    """Return the expected fill value for a given tile."""
+    round_idx = ROUND_LABELS.index(round_)
+    ch_idx = CH_LABELS.index(ch)
+    z_idx = Z_LABELS.index(z)
+    return float((((round_idx * NUM_CH) + ch_idx) * NUM_Z) + z_idx) / (NUM_ROUND * NUM_CH * NUM_Z)
+
+
+def x_coordinates(round_: int, ch: int) -> Tuple[float, float]:
+    """Return the expected physical x coordinate value for a given round/ch tuple.  Note that in
+    real life, physical coordinates are not expected to vary for different ch values.  However, for
+    completeness of the tests, we are pretending they will."""
+    return min(round_, ch) * 0.01, max(round_, ch) * 0.01
+
+
+def y_coordinates(round_: int, ch: int) -> Tuple[float, float]:
+    """Return the expected physical y coordinate value for a given round/ch tuple.  Note that in
+    real life, physical coordinates are not expected to vary for different ch values.  However, for
+    completeness of the tests, we are pretending they will."""
+    return min(round_, ch) * 0.001, max(round_, ch) * 0.001
+
+
+def z_coordinates(z: int) -> Tuple[float, float]:
+    """Return the expected physical z coordinate value for a given zlayer index."""
+    return z * 0.0001, (z + 1) * 0.0001
+
+
+class UniqueTiles(FetchedTile):
+    """Tiles where the pixel values are unique per round/ch/z."""
+    def __init__(self, fov: int, _round: int, ch: int, z: int) -> None:
+        super().__init__()
+        self._round = _round
+        self._ch = ch
+        self._z = z
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        return HEIGHT, WIDTH
+
+    @property
+    def coordinates(self) -> Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]]:
+        return {
+            Coordinates.X: x_coordinates(self._round, self._ch),
+            Coordinates.Y: y_coordinates(self._round, self._ch),
+            Coordinates.Z: z_coordinates(self._z),
+        }
+
+    @property
+    def format(self) -> ImageFormat:
+        return ImageFormat.TIFF
+
+    def tile_data(self) -> np.ndarray:
+        return np.full(
+            shape=(HEIGHT, WIDTH),
+            fill_value=fill_value(self._round, self._ch, self._z),
+            dtype=np.float32)
+
+
+def setup_imagestack() -> ImageStack:
+    """Build an imagestack with labeled indices (i.e., indices that do not start at 0 or are not
+    sequential non-negative integers).
+    """
+    collection = build_image(
+        range(1),
+        ROUND_LABELS,
+        CH_LABELS,
+        Z_LABELS,
+        tile_fetcher_factory(UniqueTiles, True),
+    )
+    tileset = list(collection.all_tilesets())[0][1]
+
+    return ImageStack.from_tileset(tileset)
+
+
+def test_labeled_indices_read():
+    """Build an imagestack with labeled indices (i.e., indices that do not start at 0 or are not
+    sequential non-negative integers).  Verify that get_slice and get_islice behaves correctly.
```

Comment:
since neither `get_slice` or `get_islice` are called in this test, can you add a bit more information here about how this works?

Reply:
`get_islice` is dead.  `verify_stack_fill` calls `get_slice`.  Updated documentation to reflect the gone-ness of `get_islice`.